core/src/replay_stage.rs

```diff
@@ -9369,4 +9370,192 @@ pub(crate) mod tests {
         assert_eq!(working_bank.slot(), good_slot);
         assert_eq!(working_bank.parent_slot(), initial_slot);
     }
+
+    #[test]
+    #[should_panic(expected = "assertion `left == right` failed")]
+    fn test_mark_slots_duplicate_confirmed() {
```

Comment:
Hmm, I think the main point of the test case would be testing that if we have [5, 6], and 5 is already marked duplicate, we should mark 6 as duplicated as well? We can probably omit the testing of panic to make it more obvious? And make the test name more descriptive maybe?

Reply:
it just tests the entire fn. [5, 6] is one of the cases see line 9419